### DISCUSSION TITLE
my master branch is not recognized what it blocked me to upload project for the first time

### DISCUSSION THREAD

**Discussion by imryn** [community/community#132227] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am trying to upload a project I developed to github for the first time, and get strange error. I have tried to find a solution to it but nothing help.
error: src refspec master does not match any
error: failed to push some refs to 'github.com:userName/projectX.git'

I used those commits:
git init 
git add .
git commit -m "first push"
git remote add origin git@github.com:userName/projectX.git

but when I try to push by using 
git push -u origin master

I got the error, I wrote you above.
also, try to use git branch to see if i get any branch, but got nothing. 
first time I get this error message. it is not the first time I upload a new project to github. 
do you have any suggestions?

I also updated github version to latest version

**Comment by Saulimedes** [comment]
It sounds like the issue might be that your local repository doesn't have a branch named master yet. By default, newer versions of Git use main as the default branch name instead of master.  Can you check if the branch exists?
